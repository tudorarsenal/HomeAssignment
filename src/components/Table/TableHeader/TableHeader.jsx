import PropTypes from 'prop-types';

import './TableHeader.scss';

const TableHeader = ({ headers, isSelectable }) => (
  <div
    className={`table-header ${isSelectable ? 'table-header-selectable' : ''}`}
  >
    {headers?.map((header) => (
      <div
        className={`table-header__item table-header__${header.toLowerCase()}`}
        key={header}
      >
        {header}
      </div>
    ))}
  </div>
);

TableHeader.propTypes = {
  headers: PropTypes.arrayOf(PropTypes.string).isRequired,
  isSelectable: PropTypes.bool,
};

export default TableHeader;
