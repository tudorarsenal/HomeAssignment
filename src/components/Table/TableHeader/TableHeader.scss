.table-header {
  display: flex;
  padding: 10px;
  border: 1px solid rgb(237, 237, 237);
  background-color: white;
  font-size: 18px;

  &__item {
    flex: 1;
  }

  &__path {
    flex: 2;
  }
}

.table-header-selectable {
  padding-left: 65px;
}

@media screen and (max-width: 800px) {
  .table-header-selectable {
    padding-left: 45px;
  }
}
