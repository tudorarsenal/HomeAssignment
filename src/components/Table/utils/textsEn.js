export const SELECTED = 'Selected';
export const NO_ROWS_SELECTED = 'None Selected';
export const DOWNLOAD_LABEL = 'Download Selected';
export const NO_DATA = 'No data';
export const NO_HEADERS = 'No Headers';
export const SELECT_ALL = 'Select all';
