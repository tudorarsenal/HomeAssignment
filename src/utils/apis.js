import { devices } from './mockData';

export const getDevices = () => devices;
