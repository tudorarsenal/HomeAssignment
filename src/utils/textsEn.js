export const ERROR_MESSAGE_DOWNLOAD =
  'Only available devices will be downloaded!';
export const NO_DEVICES_DOWNLOAD = 'No devices downloaded';
