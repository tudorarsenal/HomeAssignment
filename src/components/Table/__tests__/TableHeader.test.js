import renderer from 'react-test-renderer';

import TableHeader from '../TableHeader/TableHeader';
import { tableHeaders } from '../../../utils/mockData';

describe('TableHeader', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(<TableHeader headers={tableHeaders} isSelectable={true} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly without select all', () => {
    const tree = renderer
      .create(<TableHeader headers={tableHeaders} isSelectable={false} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
