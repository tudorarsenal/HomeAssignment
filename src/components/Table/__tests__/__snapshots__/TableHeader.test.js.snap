// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableHeader renders correctly 1`] = `
<div
  className="table-header table-header-selectable"
>
  <div
    className="table-header__item table-header__name"
  >
    Name
  </div>
  <div
    className="table-header__item table-header__device"
  >
    Device
  </div>
  <div
    className="table-header__item table-header__path"
  >
    Path
  </div>
  <div
    className="table-header__item table-header__status"
  >
    Status
  </div>
</div>
`;

exports[`TableHeader renders correctly without select all 1`] = `
<div
  className="table-header "
>
  <div
    className="table-header__item table-header__name"
  >
    Name
  </div>
  <div
    className="table-header__item table-header__device"
  >
    Device
  </div>
  <div
    className="table-header__item table-header__path"
  >
    Path
  </div>
  <div
    className="table-header__item table-header__status"
  >
    Status
  </div>
</div>
`;
